sql/core/src/main/scala/org/apache/spark/sql/streaming/ui/UIUtils.scala

```diff
@@ -57,4 +62,16 @@ private[ui] object UIUtils {
       query.exception.map(_ => "FAILED").getOrElse("FINISHED")
     }
   }
+
+  private val progressTimestampFormat = new ThreadLocal[SimpleDateFormat]() {
+    override def initialValue(): SimpleDateFormat = {
+      val format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'") // ISO8601
+      format.setTimeZone(getTimeZone("UTC"))
```

Comment:
This is useless settings because you require a time zone in the inputs:
```scala
scala> val sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
sdf: java.text.SimpleDateFormat = java.text.SimpleDateFormat@5af7aed5

scala> sdf.parse
parse   parseObject

scala> sdf.parse("2020-03-03T11:12:13.123")
java.text.ParseException: Unparseable date: "2020-03-03T11:12:13.123"
  at java.text.DateFormat.parse(DateFormat.java:366)
  ... 47 elided

scala> sdf.parse("2020-03-03T11:12:13.123Z")
res1: java.util.Date = Tue Mar 03 11:12:13 MSK 2020
```

Reply:
Sorry, I was confused. Actually, 'Z' at the end of the pattern is not a place holder for time zone. Did you do that intentionally?

Reply:
I think so. It's not for parsing but formatting, and it's always UTC, so putting `Z` at the end is corrected.

Reply:
I also checked this during the review, this timestamp string generated internally at

https://github.com/apache/spark/blob/380e8876316d6ef5a74358be2a04ab20e8b6e7ca/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala#L91-L92

So here is just parse back.